### PR TITLE
61

### PR DESCRIPTION
**Pull request by jgheerardyn**

### CODE REVIEW

**Comment by jgheerardyn** [comment]
Hi Tim
I have pulled your changes into this branch since we both touched the Explorer class

File: foundations/src/main/java/io/yields/math/framework/Explorer.java [hunk 1]
```diff
@@ -137,10 +149,20 @@ public ScoreResult getScore() {
         }
     }
 
-    public static <T, V> Explorer<V> explore(String name, String group, boolean official, Functional<T> functional, Function<T, V> operator,
+    public List<Stats> getStats() {
+        return getResults().stream().map(propertyVerifications -> {
+            if (propertyVerifications.getFunctionOutcome().isPresent()) {
```
Comment (rombautt) [inline foundations/src/main/java/io/yields/math/framework/Explorer.java]: prefer sth like this:

.map(propertyVerifications -> propertyVerifications.getFunctionOutcome()).map(functionOutcome -> statsExtractor.apply(functionOutcome)).orElse(EMPTY_STATS)

File: foundations/src/main/java/io/yields/math/framework/Stats.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+package io.yields.math.framework;
+
+import java.util.List;
+
+public interface Stats {
+
+    /**
+     * The statistics of the object summarize the functional result
```
Comment (rombautt) [inline foundations/src/main/java/io/yields/math/framework/Stats.java]: summarizing

File: foundations/src/main/java/io/yields/math/framework/Stats.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+     *
+     * @return the names of the stats
+     */
+    public List<String> getStatsNames();
+
+    /**
+     * The statistics of the object summarize the functional result
```
Comment (rombautt) [inline foundations/src/main/java/io/yields/math/framework/Stats.java]: this first line of javadoc belong in the iface level

File: foundations/src/main/java/io/yields/math/framework/junit/ExplorationContext.java [hunk 4]
```diff
@@ -50,4 +51,14 @@
         return Optional.empty();
     }
 
+    /**
+     * Maps the result of the transformed functional (operator o functional) to statistics.
+     * Example would be curve -> length of curve, average curvature, ...
+     *
+     * @return function mapping V to a set of stats.
+     */
+    default Function<V, Stats> getStatsMap() {
```
Comment (rombautt) [inline foundations/src/main/java/io/yields/math/framework/junit/ExplorationContext.java]: good use of default methods, like

**Comment by rombautt** [comment]
Reviewed

**Pull request merged**